src/msg/simple/Pipe.cc

```diff
@@ -1704,7 +1709,8 @@ void Pipe::reader()
     }
   }
 
- 
+  rcu_unregister_thread();
```

Comment:
I don't believe the RCU data structures have a problem supporting a list of that length. 

The way the code is written, we won't allow back-to-back RCU updates to pg_map without first waiting a "grace period". Meaning, the updater first makes changes and then calls synchronize_rcu(), to get indication that each thread has either blocked (went offline) or called "quiescent state". The more "online" threads which happened to be dealing with I/O or whatever, the longer the grace period could be.  But as far as I know these threads are built to very quickly hand-off I/Os to the dispatch queue, so should call rcu_quiesce in a timely fashion, hence the grace period won't be unduly long. 

That said, we could workaround a "long grace period" by calling defer_rcu() rather than synchronize_rcu(). This would (1) copy pg_map, (2) modify the copy, (3) set the pg_map pointer to the new copy (4) call defer_rcu, which would in the background free up the old pg_map out of band.  I think this would require refactoring the code to make pg_map everywhere accessed as a pointer rather than as a reference.